sql/executor.go

```diff
@@ -165,28 +165,47 @@ func (e Executor) execStmt(stmt parser.Statement, params parameters, planMaker *
 				} else {
 					switch vt := val.(type) {
 					case parser.DBool:
-						row.Values = append(row.Values, driver.Datum{BoolVal: (*bool)(&vt)})
+						row.Values = append(row.Values, driver.Datum{
+							Payload: &driver.Datum_BoolVal{BoolVal: bool(vt)},
```

Comment:
It's unfortunate that golang/protobuf doesn't have the "has" bitmask that proto2/c++ uses. It looks like if we want a union without extra allocations we'll have to add an explicit type field and make all the fields non-nullable (or we fix the `oneof` implementation do to something better).